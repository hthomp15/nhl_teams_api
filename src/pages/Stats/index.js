import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'

function Stats() {
    const [error, setError] = useState(null)
    const [isLoaded, setIsLoaded] = useState(false)
    const [teamStats, setteamStats] = useState([])
    const { id } = useParams()

    useEffect(() => {
        fetch(`https://statsapi.web.nhl.com/api/v1/teams/${id}/stats`)
            .then(res => res.json())
            .then(
                (result) => {
                    setIsLoaded(true)
                    setteamStats(result)
                },
                (error) => {
                    setIsLoaded(true)
                    setError(error)
                }
            )
    }, [])
    console.log("teamStats", teamStats)
    if (error) {
        return <div>Error: {error.message}</div>
    } else if (!isLoaded) {
        return <div>Loading...</div>
    } else {
        return (
            <div className="container">
                <h2> {teamStats.stats[0].splits[0].team.name} </h2>
                <table className="table-auto w-full">
                    <thead>
                        <tr>
                            <th className="px-4 py-2">GP</th>
                            <th className="px-4 py-2">W</th>
                            <th className="px-4 py-2">L</th>
                            <th className="px-4 py-2">PTS</th>
                            <th className="px-4 py-2">OT</th>
                            <th className="px-4 py-2">GPG</th>
                            <th className="px-4 py-2">GAPG</th>
                            <th className="px-4 py-2">SV%</th>
                            <th className="px-4 py-2">S</th>
                            <th className="px-4 py-2">SA</th>
                            <th className="px-4 py-2">S%</th>
                            <th className="px-4 py-2">FOW%</th>
                        </tr>
                    </thead>
                    <caption> Regular Season Stats </caption>
                    <tbody>
                        
                            <tr>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.gamesPlayed}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.wins}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.losses}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.pts}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.ot}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.goalsPerGame}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.goalsAgainstPerGame}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.savePctg}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.shotsPerGame}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.shotsAllowed}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.shootingPctg}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.faceOffWinPercentage}</td>
                            </tr>
                    </tbody>
                </table>
                <table className="table-auto w-full">
                    <caption> Regular Season Rankings </caption>
                    <thead>
                        <tr>
                            <th className="px-4 py-2">GP</th>
                            <th className="px-4 py-2">W</th>
                            <th className="px-4 py-2">L</th>
                            <th className="px-4 py-2">PTS</th>
                            <th className="px-4 py-2">OT</th>
                            <th className="px-4 py-2">GPG</th>
                            <th className="px-4 py-2">GAPG</th>
                            <th className="px-4 py-2">SV%</th>
                            <th className="px-4 py-2">S</th>
                            <th className="px-4 py-2">SA</th>
                            <th className="px-4 py-2">S%</th>
                            <th className="px-4 py-2">FOW%</th>
                        </tr>
                    </thead>
                 
                    <tbody>
                     
                            <tr>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.gamesPlayed}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.wins}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.losses}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.pts}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.ot}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.goalsPerGame}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.goalsAgainstPerGame}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.savePctg}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.shotsPerGame}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.shotsAllowed}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.shootingPctg}</td>
                                <td className="border px-4 py-2">{teamStats.stats[0].splits[0].stat.faceOffWinPercentage}</td>
                            </tr>
                    </tbody>
                </table>
            </div>
        )
    }
}
export default Stats;