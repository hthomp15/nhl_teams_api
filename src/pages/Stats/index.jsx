import React, { useEffect, useState } from 'react'
import { useParams, Link } from 'react-router-dom'

// Components
import TeamStatsTable from '../../components/TeamStatsTable'
import StatsKeyList from '../../components/TeamStatsTable/StatsKeyList'

function Stats(props) {
    const { tableData } = props;
    const [error, setError] = useState(null)
    const [isLoaded, setIsLoaded] = useState(false)
    const [teamStats, setTeamStats] = useState([])
    const { id } = useParams()

    // Async call to get the team stats
    useEffect(() => {
        const getTeamStats = async () => {
            try {
                const url = `https://statsapi.web.nhl.com/api/v1/teams/${id}/stats`
                const res = await fetch(url)
                console.log(res.ok)
                const data = await res.json()
                setIsLoaded(true)
                setTeamStats(data)
            } catch (error) {
                console.error(error)
                setError(error)
                setIsLoaded(true)
            }
        };
        getTeamStats()
    }, [id])
    if (error) {
        return <div>Error: {error.message}</div>
    } else if (!isLoaded) {
        return <div>Loading...</div>
    } else {

        //Passing team stats to Component
        const allStats = {
            numericalStats: teamStats.stats[0].splits[0].stat,
            leagueRanking: teamStats.stats[1].splits[0].stat,
            teamName: teamStats.stats[0].splits[0].team.name
        }
        return (
            <div className="single-team-stats-table">
                <div className="flex justify-end mx-10">
                    <Link to="/nhl_teams_api">
                        <span className="flex hover:text-white">
                            <svg xmlns="http://www.w3.org/2000/svg"
                                className="h-6 w-6 px-1" fill="none"
                                viewBox="0 0 24 24"
                                stroke="currentColor"
                                strokeWidth={2}>
                                <path strokeLinecap="round"
                                    strokeLinejoin="round"
                                    d="M10 19l-7-7m0 0l7-7m-7 7h18" />
                            </svg>
                            Back to Teams
                        </span>
                    </Link>
                </div>
                <TeamStatsTable allStats={allStats} tableData={tableData} />
            </div>
        )
    }
}
export default Stats;