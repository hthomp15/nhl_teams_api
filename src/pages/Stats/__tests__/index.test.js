import React from "react";
import { render, cleanup } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import Stats from "..";

afterEach(cleanup);

describe("Stats Component", () => {
    it("renders", () => {
        render(<Stats />);
    });
    it("matches snapshot DOM node structure", () => {
        const { asFragment } = render(<Stats />);
        expect(asFragment()).toMatchSnapshot();
    })
});