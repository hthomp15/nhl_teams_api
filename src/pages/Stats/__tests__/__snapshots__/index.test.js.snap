// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
